### ISSUE TITLE
Your help is needed for standard values for gain and threshold

### ISSUE THREAD

**Issue by Elektroarzt** [Elektroarzt/simplebus2-mqtt-bridge#9]
From experience, it was a good decision to equip the values for gain and threshold with digital potentiometers, as there is a signal dependency on every specific installation regarding attenuation on the bus wires. After extensive testing, standard parameters have crystallized for this purpose. With the values of gain at 20 and a threshold of 600mV, the up to now considered systems are operational. If you have a bridge up and running, please be so kind as to reply to this thread whether these settings also work for you, so that we can confirm these values as standards for the community. If they do not work, please write back your working values.
Both can be set easily with the new MQTT commands in the latest firmware or in the web interface (see readme).

Thanks for your support! :wave:

**Comment by alexdetsch** [comment]
For me 20/600 is working fine with the latest firmware :)

**Comment by LaurensdeGraaff** [comment]
The default is not working for me, I'm still trying to get a working config. Do you have any tips how to find them without a oscilloscope? I flashed it with esp-home, if you have a .bin file I will gladly test the values on the MQTT firmware.

**Comment by Elektroarzt** [comment]
The config values 20/600 are tested and working with the Arduino firmware. If you have flashed a 'custom' firmware, there is a completely different algorithm decoding the bus signals. So likely other standard parameters have to be chosen. The software developer could help out maybe. I will upload the current bin file shortly if you want to get back to the Arduino firmware.

**Comment by se-bastiaan** [comment]
I've recorded the values received by the bridge on the bus during a single day. The 20/600 values do allow me to receive messages, but except for a few case none of those were sent by the switchboard at the front door:

```
20:43:08 {"code": 16, "addr": 25}
20:43:07 {"code": 17, "addr": 25}
19:49:08 {"code": 21, "addr": 16}
19:48:07 {"code": 16, "addr": 16}
19:47:12 {"code": 21, "addr": 16}
19:45:42 {"code": 16, "addr": 16}
19:36:03 {"code": 21, "addr": 16}
19:34:53 {"code": 16, "addr": 16}
19:28:27 {"code": 21, "addr": 16}
17:41:13 {"code": 21, "addr": 29}
17:25:22 {"code": 18, "addr": 9}
17:25:18 {"code": 16, "addr": 9}
17:25:11 {"code": 17, "addr": 9}
17:12:09 {"code": 18, "addr": 45}
17:12:02 {"code": 16, "addr": 45}
17:11:59 {"code": 17, "addr": 45}
16:13:42 {"code": 21, "addr": 41}
16:12:34 {"code": 16, "addr": 41}
16:10:42 {"code": 16, "addr": 2}
16:10:36 {"code": 21, "addr": 2}
16:07:05 {"code": 63, "addr": 255}
16:06:54 {"code": 18, "addr": 31}
16:06:51 {"code": 16, "addr": 31}
16:06:47 {"code": 49, "addr": 31}
16:06:47 {"code": 17, "addr": 31}
15:23:21 {"code": 16, "addr": 8}
15:23:08 {"code": 63, "addr": 255}
15:22:0 {"code": 50, "addr": 8}
```

Now the past few days I have slightly decreased the voltage value, and I have so far settled on 20/300. And thanks to one of my neighbours throwing a Eurovision viewing party I have been able to test those values extensively in the past 30 minutes:

```
20:23:28 {"code": 21, "addr": 16}
20:22:40 {"code": 63, "addr": 255}
20:22:29 {"code": 16, "addr": 16}
20:22:19 {"code": 50, "addr": 16}
20:19:18 {"code": 21, "addr": 16}
20:18:31 {"code": 63, "addr": 255}
20:18:20 {"code": 16, "addr": 16}
20:18:12 {"code": 50, "addr": 16}
20:12:07 {"code": 21, "addr": 18}
20:11:16 {"code": 63, "addr": 255}
20:11:05 {"code": 16, "addr": 18}
20:10:56 {"code": 50, "addr": 18}
20:07:26 {"code": 21, "addr": 16}
20:06:20 {"code": 16, "addr": 16}
20:06:11 {"code": 50, "addr": 16}
19:53:25 {"code": 63, "addr": 255}
19:53:14 {"code": 16, "addr": 16}
```

**Comment by Elektroarzt** [comment]
Thanks for sharing your insights (and thanks to the neighbor with address 16 😂)

**Comment by LaurensdeGraaff** [comment]
Back on the official firmware. Tried the default settings and 25/100. Both nothing

```
Comp Voltage:100
Gain:25
Intercom Address:12
1
Resistor value:
4166.00Percentage:
3.84Percentage value
5write Pot5Voltage to set:
100Resistor Value:
3125.00Resistor value:
3125.00Percentage:
2.80Percentage value
4write Pot4Attempting MQTT connection...connected
```

Is it a game of testing values till you get it right or do you have tips/suggestions?
I'm living on the 4th floor, so quite high up.

**Comment by Elektroarzt** [comment]
I wrote a short [how-to](https://github.com/Elektroarzt/simplebus2-mqtt-bridge/blob/main/software/tuning%20process.md) for the tuning process. This should help.

**Comment by mgiugliano** [comment]
I would love to help. However I am complete newbie for MQTT and I am trying to use io.adafruit. I did create (from the Dashboard) a feed with the name SimpleBus but I see no heartbeat, let alone messages. Any hints to go to the next step would be immensely appreciated. 

I wonder whether the Arduino IDE serial monitor could be of help in this very initial phase of installation, although I am a bit hesitant to connect my laptop via USB to the bridge, while the device is also connected to the (35mV) simplebus lines.

**Comment by mgiugliano** [comment]
Thank you!!!!

Some progress: I disconnected the bridge from the (hw) SimpleBus and connected to my laptop. I do get Serial Monitor debug messages! I can now debug the MQTT part..

I get 
`Attempting MQTT connection...failed, rc=5 try again in 5 seconds`

This has been reported [here](https://github.com/knolleary/pubsubclient/issues/291#issuecomment-354668921) pointing to an authentication error. I replaced the password with the AIO_Key, but I get the same error. 

I have to play with the pubsubclient library and Adafruit. This has nothing to do with the simpleBus bridge, I realized.
